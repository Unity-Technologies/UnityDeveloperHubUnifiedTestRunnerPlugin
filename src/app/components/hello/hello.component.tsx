import * as React from "react";
import ShadowDOM from 'react-shadow';

export interface HelloProps { 
    target: string; 
}

export class Hello extends React.Component<HelloProps, {}> {
    public style: any = require('./hello.component.scss').toString();
    render() {
        return (
            <ShadowDOM>
                <div>
                    <style>{this.style}</style>
                    <h1>Hello {this.props.target}!</h1>
                    <p>
                        This app is made with <b>Webpack</b>, <b>Typescript</b>, <b>React</b>, <b>Sass</b>, <b>ShadowDOM</b>
                    </p>
                </div>
            </ShadowDOM>
        );
    }
}
