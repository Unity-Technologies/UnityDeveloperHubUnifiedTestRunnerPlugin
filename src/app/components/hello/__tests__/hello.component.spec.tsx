import * as React from 'react'
import { shallow } from 'enzyme';
import { Hello } from '../hello.component'

describe('HelloComponent', () => {
    it('Renders', () => {
        expect(shallow(
            <Hello target="world" />
        )).toMatchSnapshot()
    });

    it('Set\'s the title', () => {
        const helloComponent = shallow(<Hello target="world" />);

        expect(helloComponent.find('h1').text()).toEqual('Hello world!');
    });
});