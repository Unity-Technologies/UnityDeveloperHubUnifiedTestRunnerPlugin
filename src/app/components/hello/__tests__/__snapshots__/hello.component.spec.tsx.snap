// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloComponent Renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ShadowDOM
    boundaryMode="open"
    delegatesFocus={false}
    include={Array []}
    nodeName="span"
>
    <div>
        <style>
            [object Object]
        </style>
        <h1>
            Hello 
            world
            !
        </h1>
        <p>
            This app is made with 
            <b>
                Webpack
            </b>
            , 
            <b>
                Typescript
            </b>
            , 
            <b>
                React
            </b>
            , 
            <b>
                Sass
            </b>
            , 
            <b>
                ShadowDOM
            </b>
        </p>
    </div>
</ShadowDOM>,
  "nodes": Array [
    <ShadowDOM
      boundaryMode="open"
      delegatesFocus={false}
      include={Array []}
      nodeName="span"
>
      <div>
            <style>
                  [object Object]
            </style>
            <h1>
                  Hello 
                  world
                  !
            </h1>
            <p>
                  This app is made with 
                  <b>
                        Webpack
                  </b>
                  , 
                  <b>
                        Typescript
                  </b>
                  , 
                  <b>
                        React
                  </b>
                  , 
                  <b>
                        Sass
                  </b>
                  , 
                  <b>
                        ShadowDOM
                  </b>
            </p>
      </div>
</ShadowDOM>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Hello
        target="world"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Hello {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "target": "world",
        },
        "refs": Object {},
        "state": null,
        "style": "[object Object]",
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ShadowDOM
          boundaryMode="open"
          delegatesFocus={false}
          include={Array []}
          nodeName="span"
>
          <div>
                    <style>
                              [object Object]
                    </style>
                    <h1>
                              Hello 
                              world
                              !
                    </h1>
                    <p>
                              This app is made with 
                              <b>
                                        Webpack
                              </b>
                              , 
                              <b>
                                        Typescript
                              </b>
                              , 
                              <b>
                                        React
                              </b>
                              , 
                              <b>
                                        Sass
                              </b>
                              , 
                              <b>
                                        ShadowDOM
                              </b>
                    </p>
          </div>
</ShadowDOM>,
        "_debugID": 2,
        "_renderedOutput": <ShadowDOM
          boundaryMode="open"
          delegatesFocus={false}
          include={Array []}
          nodeName="span"
>
          <div>
                    <style>
                              [object Object]
                    </style>
                    <h1>
                              Hello 
                              world
                              !
                    </h1>
                    <p>
                              This app is made with 
                              <b>
                                        Webpack
                              </b>
                              , 
                              <b>
                                        Typescript
                              </b>
                              , 
                              <b>
                                        React
                              </b>
                              , 
                              <b>
                                        Sass
                              </b>
                              , 
                              <b>
                                        ShadowDOM
                              </b>
                    </p>
          </div>
</ShadowDOM>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Hello
    target="world"
/>,
}
`;
